### PR TITLE
Stores hidden state of Dev HUD

### PR DESCRIPTION
**Pull request by knod**

### CODE REVIEW

File: src/components/dev/DevHud.js [hunk 1]
```diff
@@ -81,17 +81,30 @@ const DevMenu = function ({ devProps, funcs, data, state }) {
...
-    toHideText: `^ --- Hide --- ^`,
+  constructor (props) {
+    super(props);
+
+    var toggleContent = `Hide`;
+    if (props.devProps.hidden) {
+      toggleContent = `Show`;
+    }
+
+    this.state = { visibilityToggleContent: toggleContent };
```
[the remark this answers is not in the document]
Comment (knod) [reply src/components/dev/DevHud.js]: It's true. I guess I could use a ternary the way @ethanbb demonstrated. I could live with that.

File: src/components/dev/DevHud.js [hunk 2]
```diff
@@ -81,17 +81,30 @@ const DevMenu = function ({ devProps, funcs, data, state }) {
...
-      if (prevState.hiderText === prevState.toHideText) {
-        return { hiderText: `v --- Show dev HUD --- v` };
+    var props    = this.props,
+        setDev   = props.funcs.setDev,
+        devProps = props.devProps;
+
+    var doShow = !devProps.devHidden;
+    setDev(`devHidden`, doShow);
+
+    this.setState(function (prevState) {
```
[the remark this answers is not in the document]
Comment (knod) [reply src/components/dev/DevHud.js]: Well, I guess if I weren't going to get rid of this value I could calculate the value before the `.setState()`. I suppose I was originally doing it all inside a state safe zone and just didn't adjust my thinking. Welp, getting rid of it anyway, but good catch!

**Comment by knod** [comment]
@wacii : Your points are true, but then it would go in `<App>`. I was hoping to avoid polluting `<App>` with a ton of dev hud logic and such things. I don't want `<App>` to have to think about it that much. What are your thoughts on that?

**Comment by knod** [comment]
The only reason toggling English, etc., are in `<App>` is because that's where the classes need to change in order to affect the rest of the app visually for the dev hud, and that part has been able to stay pretty well contained so far.

File: src/components/dev/DevHud.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlangevin) [reply src/components/dev/DevHud.js]: How about getting rid of the ( ) which aren't needed in the ternary?  Less to read feels easier on the eyes.

```
{ devProps.devHidden ? (
  `Show`
) : (
  `Hide`
) }

becomes 

{ 
  devProps.devHidden
  ? `Show`
  : `Hide`
}

or 

{ 
  devProps.devHidden ?
  `Show` :
  `Hide`
}
```

**Comment by jlangevin** [comment]
I can't seem to quickly put my finger on the ticket so perhaps this was a conversation on slack.  But we need a way for the public to set their language.  That's yet to be built, but I'd think we'd want the language toggle in app for that.  It would also allow us to remove the language toggle from DevHud.

File: src/components/dev/DevHud.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethanbb) [reply src/components/dev/DevHud.js]: I think we decided to try to make the multi-line ternaries we are using consistent, and when there are bigger things after `?` and `:` the parentheses do make things clearer (and help avoid an eslint conflict in some cases).

That being said, this could probably be a single-line ternary.

[the remark this answers is not in the document]
Comment (jlangevin) [reply src/components/dev/DevHud.js]: Definitely agree that it can be beneficial when they're longer.  In this case, they don't serve any real purpose.  But you raise a good point that it might make more sense to shorten this to the following which is much more purdy.  :)

```
{ devProps.devHidden ? `Show` : `Hide` }
```

[the remark this answers is not in the document]
Comment (knod) [reply src/components/dev/DevHud.js]: I can change it to the one liner. One of the things I like about the parens is that it also makes it easier to distinguish what parts are what. In this case, it isn't that complicated so maybe it's not as bad. I always have trouble spotting the `:` in there all tiny and mixed in with everything else. Another thing with the multi-line ones is that it looks more like an `if`.

**Comment by knod** [comment]
@jlangevin : That's exactly what we're planning, but we've been holding off till we have enough of the app in snippets and also translated.

**Comment by knod** [comment]
Also, this was never in an issue. It wasn't anything anyone had complained about, I was just adding my own features.

**Comment by knod** [comment]
Maybe I should make those into issues in future, though.

**Comment by knod** [comment]
@wacii , @ethanbb , @jlangevin : Is this thing ready to go? Does it need any specific changes?

**Comment by ethanbb** [comment]
Oh, you could change that ternary to a one-liner. I don't have strong feelings about it, though.

**Comment by knod** [comment]
Weird, I thought I had changed that ternary to a one-liner!

As far as `devProps` being classes, I think you're right its not intuitive. The other option I saw was to turn `devProps` into a nested object, which is then a pain to deal with when setting state. That way, there'd be a list of classes in the `devProps` or something. You're right that it does need to change some `App` classes.

If you think hiding through CSS isn't the optimal choice, I can switch it back. I think I was trying to get rid of that ternary instead of reformatting it to the style we've chosen now, but since we don't handle other hiding/showing that way, I can see how it may not be the optimal solution despite my hatred of ternaries.

**Comment by wacii** [comment]
Writing toggle english as
```js
if (english)
  setDev(false)
else
  setDev(true)
```
instead of just `setDev(!english)` hurts me a little, but you've addressed the things I pointed out.

As to the style question, my opinion on such things is still just use Prettier. So I'm no help there.

**Comment by ethanbb** [comment]
I think the way it is now with the classes does make sense actually, it was just a little unexpected at first. So I'm happy with this.

@wacii I was resistant to prettier probably because of a sunk cost fallacy since we spent a while configuring eslint, but after reading a little more about the motivation and how it works, it does sound "pretty" nice! I guess if we wanted to intentionally format our code a little differently from the "norm," e.g. to make some things a little more readable, it wouldn't be a good fit, but I like the idea of not spending undue amounts of time talking about formatting.

**Comment by knod** [comment]
The ternary and `setDev()` calls are now shortened. Does this PR need more work?
